src/qt/configurenamedialog.cpp

```diff
@@ -98,4 +111,13 @@ void ConfigureNameDialog::onDataEdited(const QString &name)
 {
     ui->dataSize->setText(tr("%1 / %2").arg(name.size()).arg(MAX_VALUE_LENGTH_UI));
     ui->dataSize->resize(ui->dataSize->fontMetrics().horizontalAdvance(ui->dataSize->text()), ui->dataSize->height());
+
+    std::string data = ui->dataEdit->text().toStdString();
+
+    if(IsValidJSON(data)){
```

Comment:
There's an edge case here. If the user enters the empty string, I'm guessing that will be marked as invalid JSON (am I correct?). It's not really ideal for the user in this scenario to be encouraged to replace it with an empty JSON object `{}` since that will waste 2 bytes of blockchain space for no reason. Can we explicitly whitelist the empty string so that it doesn't produce a warning?